charts/zookeeper/templates/zookeeper.yaml

```diff
@@ -43,10 +43,18 @@ spec:
     syncLimit: {{ .Values.config.syncLimit }}
     quorumListenOnAllIPs: {{ .Values.config.quorumListenOnAllIPs }}
   {{- end }}
+  {{- if .Values.ehemeral.enabled }}
+  ephemeral:
+    enabled: {{ .Values.ephemeral.enabled }}
+    emptydirvolumesource:
```

Comment:
add the check before this.
add the following if block before setting value for `emptydirvolumesource`
```
ephemeral:
   enabled: {{ .Values.ephemeral.enabled }}
   {{- if .Values.ephemeral.emptydirvolumesource }}
   emptydirvolumesource:
      medium: {{ .Values.ephemeral.emptydirvolumesource.medium }}
      sizeLimit: {{ .Values.ephemeral.emptydirvolumesource.sizeLimit }}
   {{- end }}
```
Also the check in front of `persistence` can be changed to be created only when `ephemeral.enabled` is set to false